lib/Parse/ParseStmt.cpp

```diff
@@ -1222,25 +1224,61 @@ validateAvailabilitySpecList(Parser &P,
     }
   }
 
-  if (ParsingMacroDefinition) {
-    if (HasOtherPlatformSpec) {
-      SourceLoc InsertWildcardLoc = P.PreviousLoc;
-      P.diagnose(InsertWildcardLoc, diag::attr_availability_wildcard_in_macro);
+  if (OtherPlatformSpecLoc != None) {
```

Comment:
Style thing: it might be cleaner to have the `if`s inside the `switch` with early exits, and you can have short `if`s, like `if (OtherPlatformSpecLoc.hasValue()) { break; }`... this would reduce nesting from 2 levels to 1 for the logic. Even if you choose to not do that, please avoid using `default:` and spell out all other cases explicitly so that we get warnings when new cases are added.